Allow sending email to BCC address when not notifying user.

Hailong opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->
Order Comments Email Copy would NOT be sent to the specified email address if the send method is set to `Bcc`, and `Notify Customer by Email` is not checked when adding a comments from backbend.

The expected behaviour is that whether the `Notify Customer by Email` is checked, an email copy is sent to the specified email. While now if the `Send Order Comments Email Copy Method` config is set to `Separate Email` it goes well. But if the config is set to `Bcc`, the email can not be sent.

The problem is that the `SenderBuilder` is used by both order notification and comments notification, so the fix is to add a flag to fix the problem for comments notification.

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
1. magento/magento2#<issue_number>: Issue title
2. ...
-->


### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. Navigate to Configuration->SALES->Sales Emails->Order Comments
2. Input email address in `Send Order Comment Email Copy To`
3. Select Bcc for `Send Order Comments Email Copy Method`.
4. Go to any order, add a new comment to the order and leave the `Notify Customer by Email` unchecked.
3. Verify the email address set in Step 2 can receive email after submitting the comment.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

Hailong committed:
Allow sending email to BCC address when not notifying user.

dmytro-ch commented on app/code/Magento/Sales/Model/Order/Email/SenderBuilder.php:
```diff
@@ -78,13 +78,14 @@ public function send()
      * @return void
      */
-    public function sendCopyTo()
+    public function sendCopyTo($toBcc = false)
```
Unfortunately, we are not allowed to change public methods signature according to the [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/).

Hailong replied:
@dmytro-ch thanks for your review. But I assume if the class is not annotated by `@api`, we can update the signature. This class is kind of an utility class used by other two classes in the same folder, which are annotated by `@api`.

Hailong replied:
@dmytro-ch I do think it's not the best idea to add this flag here too, but this the minimal change to fix this issue. But let me try with a better approach anyway.

Hailong replied:
I just moved the check of `CopyMethod` to the caller class, so that to avoid introducing the signature. It looks more clear this way.

Hailong reviewed (commented):
Updated the code to avoid modifying the method signature.

dmytro-ch commented:
Hi @Hailong,
could you please check and fix the automated tests failures?
Once the changes are applied, please squash all commits into a single one in order to cleanup history.
Thank you!

Hailong commented:
@dmytro-ch I just fixed the failures in UT. I also introduced two const in the `Magento\Sales\Model\Order\Email\Sender` class, but I'm not quite sure it's allowed in this class. If not, I will need to find some other place or replace them with plain strings.

dmytro-ch commented:
Hi @Hailong, sorry for late reply and thanks for your patience!

Unfortunately, it's not allowed to define new public constants. Please check the *Semantic Version Checker build* for reference.

Could you please also update your working branch with the latest changes from mainline?

Thank you!

Hailong committed:
Merge branch '2.3-develop' into send-comments-notify-to-bcc-address

Hailong committed:
Remove the added constants to fix the Semantic Version Checker build warnings.

Hailong commented:
Thank you @dmytro-ch , I have just removed the newly added constants. Let's see how the checking goes this time.

Hailong commented:
@dmytro-ch The build checks look good this time, can you please check?

engcom-Alfa commented:
Hi @Hailong ,
Could you, please, solve the problem with conflicting files?
Thanks!

Hailong committed:
Merge branch '2.3-develop' into send-comments-notify-to-bcc-address

Hailong commented:
Hi @engcom-Alfa , the conflict has been resolved. Thanks.

magento-engcom-team commented:
Hi @sidolov, thank you for the review.
[ENGCOM-5650](https://jira.corp.magento.com/browse/ENGCOM-5650) has been created to process this Pull Request
:eight_spoked_asterisk: @sidolov, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

engcom-Alfa commented:
:heavy_check_mark: QA Passed

engcom-Foxtrot committed:
magento/magento2#22810: Unit and static tests fix.
